backend/app/api/endpoints/knowledge_artifacts.py

```
@@ -0,0 +1,153 @@
+        ) from exc
+
+
+@router.post(
+    "/{knowledge_base_id}/artifacts",
+    response_model=KnowledgeArtifact,
+    status_code=status.HTTP_201_CREATED,
+)
+async def create_artifact(
+    knowledge_base_id: int,
+    request: KnowledgeArtifactCreate,
+    db: Session = Depends(get_db),
```

Ruff (0.15.21)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)